src/notification.h

```diff
@@ -38,20 +38,46 @@
 #include <QVariantHash>
 #include <QDBusArgument>
 
+struct NotificationData
+{
+    NotificationData();
+
+    QString appName;
+    quint32 replacesId;
+    QString appIcon;
+    QString summary;
+    QString body;
+    QHash<QString, QString> actions;
+    QVariantHash hints;
+    qint32 expireTimeout;
+};
+
+class NotificationManagerProxy;
+class NotificationPrivate;
 class Q_DECL_EXPORT Notification : public QObject
 {
     Q_OBJECT
 
 public:
-    Notification(QObject *parent = 0);
+    explicit Notification(QObject *parent = 0);
+    Notification(const NotificationData &data, QObject *parent = 0);
```

Comment:
Forget that. I was thinking of Notification(const NotificationPrivate &data, QObject *parent = 0)